lib/src/kbasesearchengine/system/TypeFileStorage.java

```diff
@@ -35,18 +35,19 @@
     // as opposed to file types for mappings
     private static final Set<String> ALLOWED_FILE_TYPES_FOR_TYPES =
             new HashSet<>(Arrays.asList(".json", ".yaml"));
-    
+    //search global type to parsing rules (spec)
```

Comment:
As far as I can tell the changes to this file should be reverted.

Reply:
From my understanding this file needs to change. The storagetypes is mapping a kbase storage type (i.e. KBaseGenomes.Genome) to a type mapping (from yaml file). Since we are now doing multiple spec, such as GenomeA.yml, GenomeB.yml, and kbase storage type is still the same, the storageTypes needs to change to an List. Otherwise GenomeB would just override GenomeA

Reply:
^that shows that duplicate storage types will override, which is the problem? We will have duplicate storage types, with different search types (index)

Reply:
Unless I'm misunderstanding what you're saying, I don't think it does show that. It shows that duplicate *search types* will override each other for a particular `StorageType`. There's only one `StorageType` per `TypeMapping` instance, but potentially many `SearchTypes`.